### ISSUE TITLE
Subpanels

### ISSUE THREAD

**Issue by Dannsei** [bit101/quicksettings#30]
Hi - quicksettings is what I was looking for - a quick and easy way to create this kind of UI. Thanks for this great work!

It would be great to be able to arrange items horizontally in a panel. For example having a group of buttons in a horizontal layout.  

From a user perspective, the intuitive way to use this requested feature would be to create a second panel (with a horizontal layout) and to add this to the main panel as a component: 
.addPanel(HorizontalPanelTitle)

Alternatively, when creating the subpanel, specify the title of the parent panel as parent. (I see the quicksettings.create method already allows parent to be specified, but I wasn’t able to get this working with another panel as parent). 

Buttons (and other components) could then be added to the subpanel in the usual way. (this horizontal subpanel could also be used to create a menu bar). The advantage of the first approach is that it would allow the subpanel to be put in a specific position within its parent panel.

In essence there would be two kinds of panels: a vertical layout panel (as currently) and a horizontal layout panel, and either kind of panel could be a component/child of another panel (and either kind could be a main panel). (Then one could have vertical panels opening from a menu bar etc.).

If adopting that approach, it would also be very cool to have a setLocalChangeHandler method to set a function that is called whenever a value in that subpanel in changed (with the subpanel title and handler as arguments). The handler might then control e.g. how selection of one checkbox component deselects some others in that subpanel.

**Comment by keith-notarize** [comment]
Good ideas. I haven't been actively using the library myself lately, and since there are no major issues being reported, I haven't been doing much with it. But eventually I'll get around to doing more with it. These are good suggestions.

**Comment by radiolondra** [comment]
@Dannsei I totally agree with you. Also expandable/collapsable groups of properties (folders) should be introduced and/or scrollbars too. If you need to handle a lot of properties (as in my case) the panel height becames too big and unusable.
Also, if I need to create more than one panel (for example, one panel for each object I create) ther's no simple way (a function) to get the panel ID and select the connected object. For example, if I have 10 or more fabric.js objects created on canvas and I need to edit the effects of 2 or more objects at the same time, it's very tricky to select the right object clicking on its own panels (ok, I did it and it works but I had to seek the parent node elements and childs...).

Anyway, the major issue actually is the space occupied by the panel with a lot of properties inside. This is a sample of what I mean... the list of needed properties is not complete.

![2017-12-18_1407](https://user-images.githubusercontent.com/20070559/34107569-14570600-e3fd-11e7-8cf6-ebe0ff151358.png)

**Comment by bit101** [comment]
Thanks for the feedback and sorry for the delay. I'll try to look into some of the issues here some time soon.

**Comment by ricardocanada** [comment]
I second the request for a horizontal button panel. For many programs I have written in other languages, that is usually my preferred layout. Especially on touch screens, it keeps the user's fingers from blocking the view of what they are trying to achieve.
